code/game/turfs/turf.dm

```diff
@@ -169,6 +171,8 @@
 		air_master.remove_from_active(src)
 
 	var/turf/W = new path(src)
+	if(!defer_change)
+		W.AfterChange()
 
 	if(istype(W, /turf/simulated))
 		W:Assimilate_Air()
```

Comment:
Alright, I've gotten this working by removing the `update_icon` call from `hide` - since all that proc does is adjust `invisibility`, a var handled by BYOND itself, the call to `update_icon` is unnecessary, and things should still function as they should after my change. All systems appear to work as they should - missions load OK, ruins load OK, no runtimes, pipes work as they do before (wrenching, manifolds, t-rays, etc.).

The last thing to do is to separate away mission areas so APCs work correctly, there.